src/components/Accordion/PAccordion.vue

```diff
@@ -1,144 +1,40 @@
-              class="p-accordion__content"
-              role="region"
-              :aria-labelledby="getHeadingId(section)"
-            >
-              <!-- this is needed because PAutoHeightTransition does not animate an elements padding correctly -->
-              <div class="p-accordion__content-padding">
-                <slot :name="getContentId(section)">
-                  <slot name="content" :section="section" :selected="isSelected(section)" />
-                </slot>
-              </div>
-            </div>
-          </template>
-        </PAutoHeightTransition>
-      </div>
-    </template>
-  </div>
+      </AccordionTrigger>
+      <AccordionContent>
+        <slot name="content" :section="section" />
+      </AccordionContent>
+    </AccordionItem>
+  </AccordionRoot>
 </template>
 
-<script lang="ts" setup generic="const T extends string">
-  import { Ref, computed, ref } from 'vue'
-  import { PAutoHeightTransition } from '@/components/AutoHeightTransition'
-  import { kebabCase } from '@/utilities'
-
-  const props = defineProps<{
-    sections: T[],
-    selected?: T | null,
-  }>()
-
-  const emit = defineEmits<{
-    (event: 'update:selected', value: T | null): void,
-  }>()
-
-  const backupSelected: Ref<T | null> = ref(null)
-
-  const selected = computed({
-    get() {
-      if (props.selected) {
-        return props.selected
-      }
-
-      if (backupSelected.value) {
-        return backupSelected.value
-      }
-
-      return null
-    },
-    set(selected: T | null) {
-      backupSelected.value = selected
-
-      emit('update:selected', selected)
-    },
+<script setup lang="ts" generic="const T extends string">
+  import {
+    AccordionRoot,
+    type AccordionRootEmits,
+    type AccordionRootProps,
+    useForwardPropsEmits
+  } from 'radix-vue'
+
+  import AccordionContent from '@/components/Accordion/PAccordionContent.vue'
+  import AccordionItem from '@/components/Accordion/PAccordionItem.vue'
+  import AccordionTrigger from '@/components/Accordion/PAccordionTrigger.vue'
+
+  // eslint-disable-next-line vue/no-unused-properties
+  const props = withDefaults(defineProps<AccordionRootProps & {
+    sections?: T[],
+  }>(), {
+    type: 'single',
+    collapsible: true,
+    sections: undefined,
   })
+  const emits = defineEmits<AccordionRootEmits>()
 
-  function toggleSection(section: T): void {
-    if (isSelected(section)) {
-      selected.value = null
-      return
-    }
-
-    selected.value = section
-  }
-
-  function isSelected(section: T): boolean {
-    return selected.value === section
-  }
-
-  function getIconClass(section: T): string | undefined {
-    if (isSelected(section)) {
-      return 'p-accordion__icon--open'
-    }
-  }
-
-  function getHeadingId(section: T): string {
-    return `${kebabCase(section)}-heading`
-  }
-
-  function getContentId(section: T): string {
-    return `${kebabCase(section)}-content`
-  }
+  const forwarded = useForwardPropsEmits(props, emits)
```

Comment:
Resolved sync. This is really only needed for emits. And it's because when you use `defineEmits` that removes any matching event handlers from attrs similar to how any props you define are removed from attrs. So this composition gives them back to you in the same `'onUpdate:modelValue': f()` format that vue expects when you bind an event handler without the `@`.